### PR TITLE
Docs: IAM plugin json field

### PR DESCRIPTION
**Pull request by gamab**
**What this PR does / why we need it**:

This PR documents the new IAM field that allows a plugin to request Grafana for a custom service account.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/5992

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/metadata.md [hunk 1]
```diff
@@ -79,6 +80,29 @@ Grafana Enterprise specific features.
 | ------------------------- | ------- | -------- | ------------------------------------------------------------------- |
 | `healthDiagnosticsErrors` | boolean | No       | Enable/Disable health diagnostics errors. Requires Grafana >=7.5.5. |
 
+## iam
+
+Identity and Access Management section. Grafana will read this section and initialize a service account for the plugin, with a tailored set of
+[Grafana RBAC permissions](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/custom-role-actions-scopes/#rbac-permissions-actions-and-scopes).
+Then its bearer token will be share to the plugin backend using the `GF_PLUGIN_APP_CLIENT_SECRET` environment variable.
+
+Requires Grafana >= 10.3.0. Currently behind a feature toggle `externalServiceAccounts`.
```
Comment (gamab) [inline docusaurus/docs/metadata.md:89]: Should I wait for the feature toggle to be removed?

**Review by andresmgot — COMMENTED** [review]
I don't know what's the policy but shouldn't this be published after removing the feature flag?

**Comment by gamab** [comment]
> I don't know what's the policy but shouldn't this be published after removing the feature flag?

Good question, I had the same... I'm going back and forth on this. Because we are going to roll this out to cloud before removing the feature flag and hence plugin developers will already be able to use it there.

File: docusaurus/docs/metadata.md [hunk 2]
```diff
@@ -33,12 +33,13 @@ The `plugin.json` file is required for all plugins. When Grafana starts, it scan
...
 | `executable`         | string                        | No       | The first part of the file name of the backend component executable. There can be multiple executables built for different operating system and architecture. Grafana will check for executables named `<executable>_<$GOOS>_<lower case $GOARCH><.exe for Windows>`, e.g. `plugin_linux_amd64`. Combination of $GOOS and $GOARCH can be found here: https://golang.org/doc/install/source#environment. |
+| `iam`                | [object](#iam)                | No       | Initialize a service account for the plugin, with a tailored set of RBAC permissions. You may get its bearer token through the `GF_PLUGIN_APP_CLIENT_SECRET` environment variable.                                                                                                                                                                                                                      |
```
Comment (andresmgot) [inline docusaurus/docs/metadata.md:36]: I wouldn't add too many details here:

```suggestion
| `iam`                | [object](#iam)                | No       | Initialize a service account for the plugin, with a tailored set of RBAC permissions. |
```

File: docusaurus/docs/metadata.md (hunk 1 above)
Comment (andresmgot) [reply docusaurus/docs/metadata.md:89]: Maybe we can link here to the example that uses it?

**Review by andresmgot — COMMENTED** [review]
Mostly LGTM, I'll let @sympatheticmoose the decission on when to publish this, at the very least, we'd need to wait until 10.3.0 is released I guess.

File: docusaurus/docs/metadata.md [hunk 3]
```diff
@@ -79,6 +80,29 @@ Grafana Enterprise specific features.
 | ------------------------- | ------- | -------- | ------------------------------------------------------------------- |
 | `healthDiagnosticsErrors` | boolean | No       | Enable/Disable health diagnostics errors. Requires Grafana >=7.5.5. |
 
+## iam
+
+Identity and Access Management section. Grafana will read this section and initialize a service account for the plugin, with a tailored set of
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:85]:
```suggestion
Grafana reads the Identity and Access Management section and initializes a service account for the plugin, with a tailored set of
```

File: docusaurus/docs/metadata.md [hunk 4]
```diff
@@ -79,6 +80,29 @@ Grafana Enterprise specific features.
...
+[Grafana RBAC permissions](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/custom-role-actions-scopes/#rbac-permissions-actions-and-scopes).
+Then its bearer token will be shared with the plugin backend using the `GF_PLUGIN_APP_CLIENT_SECRET` environment variable.
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:87]:
```suggestion
After IAM is read, Grafana shares its bearer token with the plugin backend using the `GF_PLUGIN_APP_CLIENT_SECRET` environment variable.
```

File: docusaurus/docs/metadata.md [hunk 5]
```diff
@@ -79,6 +80,29 @@ Grafana Enterprise specific features.
...
+
+Requires Grafana >= 10.3.0. Currently behind a feature toggle `externalServiceAccounts`.
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:89]:
```suggestion
Requires Grafana version 10.3.0 or later. Currently, this is behind the `externalServiceAccounts` feature toggle.
```

**Review by josmperez — CHANGES_REQUESTED** [review]
Just a few editorial recommendations.

**Comment by gamab** [comment]
Thanks for the valuable feedback! :)

**Review by andresmgot — APPROVED** [review]
Content LGTM

**Review by mckn — COMMENTED** [review]
LGTM!

**Review by sympatheticmoose — APPROVED** [review]
🚀

File: docusaurus/docs/metadata.md [hunk 6]
```diff
@@ -79,6 +80,31 @@ Grafana Enterprise specific features.
...
 
+## iam
+
+Grafana reads the Identity and Access Management section and initializes a service account for the plugin, with a tailored set of
+[Grafana RBAC permissions](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/custom-role-actions-scopes/#rbac-permissions-actions-and-scopes).
+Grafana will share the service account's bearer token with the plugin backend using the `GF_PLUGIN_APP_CLIENT_SECRET` environment variable.
+
+Requires Grafana version 10.3.0 or later. Currently, this is behind the `externalServiceAccounts` feature toggle.
+
+To try this feature out, follow the example provided [here](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/app-with-service-account/README.md).
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:91]:
```suggestion
To try this feature out, follow this [example](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/app-with-service-account/README.md).
```

File: docusaurus/docs/metadata.md [hunk 7]
```diff
@@ -79,6 +80,31 @@ Grafana Enterprise specific features.
...
+
+### Properties
+
+| Property      | Type                    | Required | Description                                            |
+| ------------- | ----------------------- | -------- | ------------------------------------------------------ |
+| `permissions` | [object](#permission)[] | No       | RBAC permissions the plugin requires to query Grafana. |
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:97]:
```suggestion
| `permissions` | [object](#permission)[] | No       | Required RBAC permissions to query Grafana. |
```

File: docusaurus/docs/metadata.md [hunk 8]
```diff
@@ -79,6 +80,31 @@ Grafana Enterprise specific features.
...
+
+### permissions
+
+#### Properties
+
+| Property | Type   | Required | Description                |
+| -------- | ------ | -------- | -------------------------- |
+| `action` | string | **Yes**  | Action, e.g: `teams:read`. |
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:105]:
```suggestion
| `action` | string | **Yes**  | Action, for example: `teams:read`. |
```

Comment (josmperez) [inline docusaurus/docs/metadata.md:105]:
```suggestion
| `action` | string | **Yes**  | Action, for example: `teams:read`. |
```

**Review by josmperez — APPROVED** [review]
LGTM, just a few nits.

**Comment by gamab** [comment]
Seems like I miss a label? Which one should I use? :slightly_smiling_face:

**Pull request merged**